File: helfomat-infrastructure-google-maps/src/main/java/de/helfenkannjeder/helfomat/infrastructure/google/maps/GoogleMapsConfiguration.java
package de.helfenkannjeder.helfomat.infrastructure.google.maps;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

/**
 * @author Valentin Zickner
 */
@Component
@ConfigurationProperties
public class GoogleMapsConfiguration {

    private String apiKey;

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }
}

File: helfomat-infrastructure-google-maps/src/main/kotlin/de/helfenkannjeder/helfomat/infrastructure/google/maps/GoogleMapsConfiguration.kt
package de.helfenkannjeder.helfomat.infrastructure.google.maps

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

/**
 * @author Valentin Zickner
 */
@ConstructorBinding
@ConfigurationProperties("googlemaps")
data class GoogleMapsConfiguration(
    var apiKey: String
)